synapse/handlers/federation.py

```diff
@@ -2503,7 +2508,17 @@ async def _check_for_soft_fail(
         try:
             event_auth.check(room_version_obj, event, auth_events=current_auth_events)
         except AuthError as e:
-            logger.warning("Soft-failing %r because %s", event, e)
+            logger.warning(
+                "Soft-failing %r (from %s) because %s",
+                event,
+                e,
+                origin,
+                extra={
+                    "room_id": event.room_id,
+                    "mxid": event.sender,
+                    "hs": origin,
+                },
```

Comment:
For another example of `extra` see [`tests/logging/test_terse_json.py#L64-L74`](https://github.com/matrix-org/synapse/blob/13577aa55ebe6087e8b813c0643bbb53148e9510/tests/logging/test_terse_json.py#L64-L74) which adds extra fields to the structured logging objects.

"The fourth keyword argument is `extra` which can be used to pass a dictionary which is used to populate the `__dict__` of the `LogRecord`", https://docs.python.org/3/library/logging.html#logging.Logger.debug